lib/qml/lib/demo.py

```diff
@@ -287,6 +287,40 @@ def _build_demo(
             pre=dev,
         )
 
+        # If dev, we need to re-install the latest Catalyst, then Lightning, then PennyLane
+        # in that order, regardless of conflicts/warnings. 
+        if dev:
+            # Catalyst
+            cmds.pip_install(
+                build_venv.python,
+                "--upgrade",
+                "--extra-index-url",
+                "https://test.pypi.org/simple/",
+                "PennyLane-Catalyst",
+                use_uv=False,
+                quiet=False,
+                pre=True,
+            )
+            # Lightning
+            cmds.pip_install(
+                build_venv.python,
+                "--upgrade",
+                "--extra-index-url",
+                "https://test.pypi.org/simple/",
+                "PennyLane-Lightning",
+                use_uv=False,
+                quiet=False,
+                pre=True,
+            )
+            # PennyLane
+            cmds.pip_install(
+                build_venv.python,
+                "--upgrade",
+                "git+https://github.com/PennyLaneAI/pennylane.git#egg=pennylane",
+                use_uv=False,
+                quiet=False,
+            )
```

Comment:
```suggestion
if dev:
    packages = {
        "PennyLane-Catalyst": "https://test.pypi.org/simple/",
        "PennyLane-Lightning": "https://test.pypi.org/simple/"
    }

    # Install packages from TestPyPI
    for package, index_url in packages.items():
        cmds.pip_install(
            build_venv.python,
            "--upgrade",
            "--extra-index-url",
            index_url,
            package,
            use_uv=False,
            quiet=False,
            pre=True,
        )

    # Install PennyLane from Git
    cmds.pip_install(
        build_venv.python,
        "--upgrade",
        "git+https://github.com/PennyLaneAI/pennylane.git#egg=pennylane",
        use_uv=False,
        quiet=False,
    )
```

:major-loaf: nit: 

We could remove some repetitive code here by looping over testpypi reqs.

Reply:
sorry formatting isn. gh suggestions is awful 😢

Reply:
I've kept them separate in case we want to change this later. Either way is fine by me.